driver-core/src/main/java/com/datastax/driver/core/EventDebouncer.java

```diff
@@ -151,6 +155,7 @@ private void completeAllPendingFutures() {
 
         if (state == State.RUNNING) {
             int count = eventCount.get();
+            int maxPendingEvents = maxPendingEvents();
```

Comment:
Hmm right. It could be a concern because the scheduled delivery could happen before your 51th event is added to the queue, in which case it would remain in the queue until the next event is received. I'm thinking about changing the if condition to `count >= maxPendingEvents`. That would certainly trigger too many immediate deliveries, but on the other hand, no event would ever remain in the queue without triggering a delivery (delayed or immediate).